crates/red_knot_python_semantic/src/types/unpacker.rs

```diff
@@ -121,7 +121,7 @@ impl<'db> Unpacker<'db> {
                     if let Some(tuple_ty) = ty.into_tuple() {
                         let tuple_ty_elements = self.tuple_ty_elements(target, elts, tuple_ty);
 
-                        match elts.len().cmp(&tuple_ty_elements.len()) {
+                        let invalid_unpacking = match elts.len().cmp(&tuple_ty_elements.len()) {
```

Comment:
I think the value of this is not exactly what it represents because in `tuple_ty_elements` method, we would return a vector of unknowns that's exactly the same length as the number of targets but that's mainly because the error message is a little bit different in that case.

I don't think there's any action item here except that I'd probably rename this variable to something like `length_mismatch`.